sw/device/silicon_creator/lib/cert/cert.c

```diff
@@ -78,9 +82,10 @@ rom_error_t cert_x509_asn1_check_serial_number(
   for (size_t i = 0; i < kCertX509Asn1SerialNumberSizeIn32BitWords; ++i) {
     curr_sn_word = 0;
     for (size_t j = 0; j < sizeof(uint32_t); ++j) {
-      curr_sn_word |=
-          (uint32_t)cert_bytes[sn_bytes_offset + (i * sizeof(uint32_t)) + j]
-          << (8 * j);
+      size_t idx = (i * sizeof(uint32_t)) + j;
+      if (idx < asn1_integer_length) {
+        curr_sn_word |= (uint32_t)cert_bytes[sn_bytes_offset + idx] << (8 * j);
+      }
```

Comment:
what happens if the serial number is 19 bytes? is the comparison off by one byte then?

Maybe this would be cleaner if we just copied the serial number bytes out of cert_bytes into a separate fixed size 32-byte buffer with the bytes in LE order and the proper 0 padding in the MSB if needed. The comparison can then be a simple byte-by-byte  check with `expected_sn_bytes`. WDYT?